homeassistant/components/tellstick/sensor.py

```diff
@@ -74,25 +72,27 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     sensors = []
     datatype_mask = config.get(CONF_DATATYPE_MASK)
 
+    if config[CONF_ONLY_NAMED]:
+        named_sensors = {
+            named_sensor[CONF_ID]: named_sensor[CONF_NAME]
+            for named_sensor in config[CONF_ONLY_NAMED]}
+
     for tellcore_sensor in tellcore_lib.sensors():
         if not config[CONF_ONLY_NAMED]:
             sensor_name = str(tellcore_sensor.id)
         else:
-            sensor_name = None
-            for named_sensor in config[CONF_ONLY_NAMED]:
-                if named_sensor[CONF_ID] == tellcore_sensor.id:
-                    sensor_name = named_sensor[CONF_NAME]
-
-        if sensor_name is None:
-            continue
-        else:
-            for datatype in sensor_value_descriptions:
-                if datatype & datatype_mask:
-                    if tellcore_sensor.has_value(datatype):
-                        sensor_info = sensor_value_descriptions[datatype]
-                        sensors.append(TellstickSensor(
-                            sensor_name, tellcore_sensor,
-                            datatype, sensor_info))
+            if tellcore_sensor.id in named_sensors:
```

Comment:
Invert this and make it a guard clause where we continue if true. Then we don't need the else case and can outdent below.

Reply:
Hmm, yes, that is more effecient!
What is best practice, should i still indent continue on a new line below the condition like this:

```
   for tellcore_sensor in tellcore_lib.sensors():
        if not config[CONF_ONLY_NAMED]:
            sensor_name = str(tellcore_sensor.id)
        else:
            if not tellcore_sensor.id in named_sensors:
                continue
            sensor_name = named_sensors[tellcore_sensor.id]
```

Reply:
Yes, but it's nicer to write `if key not in container` than `if not key in container`.

Reply:
Yep, Hound caught it as well.  fixed now. =)
Thanks for your patience, I learn a lot, and love the attention to detail!